lms/djangoapps/verify_student/tests/test_api.py

```diff
@@ -41,3 +49,140 @@ def test_send_approval(self, use_ace):
+
+        self.assertEqual(verification_attempt.user, self.user)
+        self.assertEqual(verification_attempt.name, 'Tester McTest')
+        self.assertEqual(verification_attempt.status, VerificationAttemptStatus.created)
+        self.assertEqual(verification_attempt.expiration_datetime, None)
+
+
+@ddt.ddt
+class UpdateVerificationAttempt(TestCase):
+    """
+    Test cases for the update_verification_attempt API method.
+    """
+
+    def setUp(self):
+        super().setUp()
+
+        self.user = UserFactory.create()
+        self.attempt = VerificationAttempt(
+            user=self.user,
+            name='Tester McTest',
+            status=VerificationAttemptStatus.created,
+            expiration_datetime=datetime(2024, 12, 31, tzinfo=timezone.utc)
+        )
+        self.attempt.save()
+
+    @ddt.data(
+        ('Tester McTest', VerificationAttemptStatus.pending, datetime(2024, 12, 31, tzinfo=timezone.utc)),
+        ('Tester McTest2', VerificationAttemptStatus.approved, datetime(2025, 12, 31, tzinfo=timezone.utc)),
+        ('Tester McTest3', VerificationAttemptStatus.denied, datetime(2026, 12, 31, tzinfo=timezone.utc)),
+    )
+    @ddt.unpack
+    def test_update_verification_attempt(self, name, status, expiration_datetime):
+        update_verification_attempt(
+            attempt_id=self.attempt.id,
+            name=name,
+            status=status,
+            expiration_datetime=expiration_datetime,
+        )
+
+        verification_attempt = VerificationAttempt.objects.get(id=self.attempt.id)
+
+        # Values should change as a result of this update.
+        self.assertEqual(verification_attempt.user, self.user)
+        self.assertEqual(verification_attempt.name, name)
+        self.assertEqual(verification_attempt.status, status)
+        self.assertEqual(verification_attempt.expiration_datetime, expiration_datetime)
+
+    def test_update_verification_attempt_none_values(self):
+        update_verification_attempt(
+            attempt_id=self.attempt.id,
+            name=None,
+            status=None,
+            expiration_datetime=None,
+        )
+
+        verification_attempt = VerificationAttempt.objects.get(id=self.attempt.id)
+
+        # Values should not change as a result of the values passed in being None.
+        self.assertEqual(verification_attempt.user, self.user)
+        self.assertEqual(verification_attempt.name, self.attempt.name)
+        self.assertEqual(verification_attempt.status, self.attempt.status)
+        self.assertEqual(verification_attempt.expiration_datetime, self.attempt.expiration_datetime)
+
+    def test_update_verification_attempt_not_found(self):
+        self.assertRaises(
+            VerificationAttempt.DoesNotExist,
+            update_verification_attempt,
+            attempt_id=999999,
+            status=VerificationAttemptStatus.approved,
+        )
+
+    # These are statuses used in edx-name-affirmation's VerifiedName model and persona-integration's unique
```

Comment:
I think we should remove this comment, especially about `persona-integration`, which we shouldn't mention in the `edx-platform`, since it's a private plugin.

I don't think you need to explain where these statuses come from - testing invalid statuses is a normal thing to do, so it's justified as is!